clerk/sessions.go

```diff
@@ -15,10 +15,47 @@ type Session struct {
 	AbandonAt    int64  `json:"abandon_at"`
 }
 
-func (s *SessionsService) ListAll() ([]Session, error) {
+type ListAllSessionsParams struct {
+	Limit    *int
+	Offset   *int
+	ClientID *string
+	UserID   *string
+	Status   *SessionStatus
+}
+
+type SessionStatus string
+
+const (
+	STATUS_ABANDONED SessionStatus = "abandoned"
+	STATUS_ACTIVE    SessionStatus = "active"
+	STATUS_ENDED     SessionStatus = "ended"
+	STATUS_EXPIRED   SessionStatus = "expired"
+	STATUS_REMOVED   SessionStatus = "removed"
+	STATUS_REPLACED  SessionStatus = "replaced"
+	STATUS_REVOKED   SessionStatus = "revoked"
+)
+
+func (s *SessionsService) ListAll(params ListAllSessionsParams) ([]Session, error) {
```

Comment:
🔧 i would expect the params to be optional to avoid making this a BREAKING CHANGE.
```
ListAll(params *ListAllSessionsParams) 
```

Reply:
As @dimkl mentioned, simply adding the params here would be a breaking change. We're not ready for a major version bump yet unfortunately.

Even adding them as a pointer would be a breaking change, so we can't do `ListAll(p *ListAllSessionsParams)`.

I think the correct way to approach this would be to add a separate method on the SessionsService. Something like

```go
func (s *SessionService) ListAllWithFiltering(params ListAllWithFilteringSessionsParams) {}
```

@petergeorgas What do you think about that? Open to any other suggestion of course. The important thing is to be backwards compatible.

Reply:
Good point! Just made the changes!